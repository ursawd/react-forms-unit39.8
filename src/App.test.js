import React from "react";
import { render, fireEvent, queryByText } from "@testing-library/react";
import App from "./App";
//smoke test
test("renders without crashing", function () {
  render(<App />);
});

//snapshot
test("matches snapshot", function () {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test("has title", function () {
  const { queryByText } = render(<App />);
  expect(queryByText("Todo List")).toBeInTheDocument();
});
