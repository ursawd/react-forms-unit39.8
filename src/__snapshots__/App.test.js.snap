// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <h1>
        Todo List
      </h1>
      <form>
        <input
          id="task"
          name="task"
          placeholder="Enter new ToDo here..."
          type="Text"
          value=""
        />
        <button
          class="input-btn"
          type="submit"
        >
          Add New ToDo
        </button>
      </form>
      <div>
        <button
          class="btn"
          type="button"
        >
          ❌
        </button>
        <h3
          class="todo-item"
        >
          Finish App
        </h3>
      </div>
      <div>
        <button
          class="btn"
          type="button"
        >
          ❌
        </button>
        <h3
          class="todo-item"
        >
          Finish Testing
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
